submissions/UA-girl/task_friends_app/js/app.js

```diff
@@ -0,0 +1,300 @@
+    }
+
+    compareDateRegistration(other) {
+        return Date.parse(this.registered.date) - Date.parse(other.registered.date);
+    }
+}
+
+class App {
+    constructor(container) {
+        this.container = container;
+        this.filterStates = {
+            gender: null,
+            age: null
+        };
+        this.sortingStates = {
+            name: null,
+            registration: null,
+            age: null
+        };
+        this.cards = [];
+        this.currentPage = 0;
+    }
+
+    restoreFromFilter() {
+        this.currentPage = 0;
+        this.cards = cards.slice();
+        for (let [filter, state] of Object.entries(this.filterStates)) {
+            if (state) {
+                switch (filter) {
+                    case 'gender':
+                        this.genderFilter(state, false);
+                        break;
+                    case 'age':
+                        this.ageFilter(state || MAX_AGE);
+                        break;
+                }
+            }
+        }
+        for (let [filter, state] of Object.entries(this.sortingStates)) {
+            if (state) {
+                this.sortCards(state, filter);
+            }
+        }
+    }
+
+    resetFilterState(updateGender = false) {
+        this.filterStates.age = null;
+        if (updateGender) {
+            this.filterStates.gender = null
+        }
+    }
+
+    resetSortingState() {
+        this.sortingStates = {
+            name: null,
+            registration: null,
+            age: null
+        };
+    }
+
+    genderFilter(gender, wasToggled = true) {
+        this.filterStates.gender = gender;
+        if (wasToggled) {
+            this.restoreFromFilter();
+        } else {
+            this.cards = this.cards.filter(person => person.gender === gender);
+        }
+    }
+
+    ageFilter(maxAge) {
+        document.querySelector('.label-range').innerHTML = `Max age: ${maxAge}`;
+        if (this.filterStates.age < +maxAge) {
+            this.filterStates.age = +maxAge;
+            this.restoreFromFilter();
+        } else {
+            this.filterStates.age = +maxAge;
+            this.cards = this.cards.filter(person => person.age <= +maxAge);
+        }
+    }
+
+    sortCards(value, filterName) {
+        this.resetSortingState();
+        switch (filterName) {
+            case 'age':
+                this.cards.sort((person1, person2) => person1.compareAge(person2));
+                break;
+            case 'name':
+                this.cards.sort((person1, person2) => person1.compareName(person2));
+                break;
+            case 'registration':
+                this.cards.sort((person1, person2) => person1.compareDateRegistration(person2));
+        }
+        this.sortingStates[filterName] = value;
+        if (value === 'down') {
+            this.cards.reverse()
+        }
+    }
+
+    nameFilter(name) {
+        name = name.trim().toLowerCase();
```

Comment:
I'd recommend creating a new variable for the `name` here. Because it would be confused to find updated values of variables